packages/sanity/src/desk/comments/src/context/selected-path/CommentsSelectedPathProvider.tsx

```diff
@@ -0,0 +1,28 @@
+import React, {useMemo, useState} from 'react'
+import {CommentsSelectedPathContext} from './CommentsSelectedPathContext'
+import {CommentsSelectedPath, CommentsSelectedPathContextValue} from './types'
+
+interface CommentsSelectedPathProviderProps {
+  children: React.ReactNode
+}
+
+export const CommentsSelectedPathProvider = React.memo(function CommentsSelectedPathProvider(
+  props: CommentsSelectedPathProviderProps,
+) {
+  const {children} = props
+  const [selectedPath, setSelectedPath] = useState<CommentsSelectedPath>(null)
```

Comment:
(To preface: this new behaviour feels so much more intuitive, well done!)

As `setSelectedPath` receives an object literal, any `setSelectedPath` call will always trigger a _full_ re-render of the document pane, regardless of whether any path values (`fieldPath`, `threadId` etc) have changed or not.

This is especially noticeable when clicking between different threads within a field path. On local, this previously used to feel _instant_ but now there's a definite hitch, which is likely due to the entire document form re-rendering. 

This issue is also exacerbated by the fact that `setSelectedPath` is being called whenever you click _anywhere_ in the comment. Making a text selection in a comment will trigger two renders, which we probably don't want!

Rather than passing the `useState` set function to context – perhaps we can pass a custom callback which does a shallow comparison before setting internal state?